data/json/effects_on_condition/npc_eocs/appearance_eocs.json

```diff
@@ -0,0 +1,325 @@
+              "or": [ { "u_has_trait": "hair_short" }, { "u_has_trait": "hair_short_no_fringe" }, { "u_has_trait": "hair_short_over_eye" } ]
+            },
+            "then": [
+              { "u_lose_trait": "hair_short" },
+              { "u_lose_trait": "hair_short_no_fringe" },
+              { "u_lose_trait": "hair_short_over_eye" },
+              { "run_eoc_with": "reset_natural_hair_color", "variables": { "trait_id": "hair_medium" } }
+            ],
+            "else": {
+              "if": { "or": [ { "u_has_trait": "hair_medium" } ] },
+              "then": [
+                { "u_lose_trait": "hair_medium" },
+                { "run_eoc_with": "reset_natural_hair_color", "variables": { "trait_id": "hair_long" } }
+              ],
+              "else": {
+                "if": { "u_has_trait": "hair_mohawk" },
+                "then": [
+                  { "u_lose_trait": "hair_mohawk" },
+                  { "run_eoc_with": "reset_natural_hair_color", "variables": { "trait_id": "hair_long_mohawk" } }
+                ],
+                "else": {
+                  "if": {
+                    "or": [
+                      { "u_has_trait": "hair_long" },
+                      { "u_has_trait": "hair_long_mohawk" },
+                      { "u_has_trait": "hair_mullet" },
+                      { "u_has_trait": "hair_long_over_eye" }
+                    ]
+                  },
+                  "then": [
+                    { "u_lose_trait": "hair_long" },
+                    { "u_lose_trait": "hair_mullet" },
+                    { "u_lose_trait": "hair_long_over_eye" },
+                    { "u_lose_trait": "hair_long_mohawk" },
+                    { "run_eoc_with": "reset_natural_hair_color", "variables": { "trait_id": "hair_messy" } }
+                  ],
+                  "else": {
+                    "if": { "or": [ { "u_has_trait": "hair_messy" } ] },
+                    "then": [
+                      { "u_lose_trait": "hair_messy" },
+                      { "run_eoc_with": "reset_natural_hair_color", "variables": { "trait_id": "hair_very_long" } }
+                    ],
+                    "else": {
+                      "if": { "or": [ { "u_has_trait": "hair_very_long" } ] },
+                      "then": [
+                        { "u_lose_trait": "hair_very_long" },
+                        {
+                          "run_eoc_with": "reset_natural_hair_color",
+                          "variables": { "trait_id": "hair_extremely_long" }
+                        }
+                      ],
+                      "else": {
+                        "if": { "or": [ { "u_has_trait": "hair_extremely_long" } ] },
+                        "then": [
+                          { "u_lose_trait": "hair_extremely_long" },
+                          {
+                            "run_eoc_with": "reset_natural_hair_color",
+                            "variables": { "trait_id": "hair_body_length" }
```

Comment:
additionally, you can make a variable "current hair length", that just measure how long your hair should be, and a single EoC that does 
```
{
  if: hair length > 5
  then: give this trait
}
{
  if: hair length > 8
  then: give that trait
},
{
  if: hair length > 11
  then: give those trait
}
```
so you don't have this giant ladder of `if-then-else if`

Reply:
I'll probably save that for a follow-up PR, but I'll keep this in mind. So far, I've gotten the health dependency set up.